source/time_stepping/convection_time_step.cc

```diff
@@ -36,13 +36,19 @@ namespace aspect
       FEValues<dim> fe_values (this->get_mapping(),
                                this->get_fe(),
                                quadrature_formula,
-                               update_values);
+                               update_values   |
+                               update_gradients |
+                               update_quadrature_points |
+                               update_JxW_values);
 
       const unsigned int n_q_points = quadrature_formula.size();
 
       std::vector<Tensor<1,dim>> velocity_values(n_q_points);
       std::vector<Tensor<1,dim>> fluid_velocity_values(n_q_points);
 
+      MaterialModel::MaterialModelInputs<dim> in(fe_values.n_quadrature_points, this->n_compositional_fields());
+      MaterialModel::MaterialModelOutputs<dim> out(fe_values.n_quadrature_points, this->n_compositional_fields());
+      MaterialModel::MeltOutputs<dim> *fluid_out = out.template get_additional_output<MaterialModel::MeltOutputs<dim>>();
```

Comment:
at this point you have not yet asked the material model to create additional named outputs. See [here](https://github.com/geodynamics/aspect/blob/0d485a6bb605e2f4d753b37a9de3a760b8e4f924/source/simulator/assembly.cc#L935) for what you need to do.

Reply:
I was still not able to get this to not be a `nullptr` unfortunately. I tried placing the location of this line before/after running evalute, or before and after filling the additional_output, but it seems that it is still a `nullptr` no matter what I try